plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -3339,4 +3340,43 @@ public void testOptimizeParameterValidation()
                 "ALTER TABLE nation EXECUTE OPTIMIZE (file_size_threshold => '33s')",
                 "\\QUnable to set procedure property 'file_size_threshold' to ['33s']: Unknown unit: s");
     }
+
+    @Test
+    public void testRepartitioningIsForcedDuringOptimize()
```

Comment:
We should have tests for OPTIMIZE with partitioned tables.
Actually, this may be the test. Just rename it.
The "forced repartitioning" will then be an inline comment where the assertion is done